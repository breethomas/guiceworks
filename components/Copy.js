// @flow
import React from 'react'
import css, { media2 } from '../styles/css'
import SansRegular from './SansRegular'
import SansThin from './SansThin'

type Props = {
  isSmall?: boolean,
  isThin?: boolean,
  tag?: string,
}

const style = css(
  {
    margin: '1em 0',
    fontSize: 16,
    lineHeight: 1.3,
  },
  media2({
    fontSize: 20,
  }),
)

const styleSmall = css(
  {
    margin: '1em 0',
    fontSize: 12,
    lineHeight: 1.3,
  },
  media2({
    fontSize: 14,
  }),
)

const Copy = ({ isSmall, isThin, tag, ...props }: Props) => {
  const textProps = { className: isSmall ? styleSmall : style, tag }
  return isThin ?
    <SansThin {...textProps} {...props} />
  :
    <SansRegular {...textProps} {...props} />
}

Copy.defaultProps = {
  isSmall: false,
  isThin: false,
  tag: 'p',
}

export default Copy

