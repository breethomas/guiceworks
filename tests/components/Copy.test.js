import React from 'react'
import renderer from 'react-test-renderer'
import Copy from '../../components/Copy'

describe('Copy', () => {
  it('shows the correct snapshot for the Copy component tree (default)', () => {
    const component = <Copy>Copy</Copy>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot for the Copy component tree (isSmall)', () => {
    const component = <Copy isSmall>Copy</Copy>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot for the Copy component tree (isThin)', () => {
    const component = <Copy isThin>Copy</Copy>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot for the Copy component tree (tag)', () => {
    const component = <Copy tag="span">Copy</Copy>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
