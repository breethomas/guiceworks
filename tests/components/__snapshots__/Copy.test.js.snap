// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy shows the correct snapshot for the Copy component tree (default) 1`] = `
<p
  className="css-6p63zg css-1eyhu5m"
>
  Copy
</p>
`;

exports[`Copy shows the correct snapshot for the Copy component tree (isSmall) 1`] = `
<p
  className="css-6p63zg css-1jmoywn"
>
  Copy
</p>
`;

exports[`Copy shows the correct snapshot for the Copy component tree (isThin) 1`] = `
<p
  className="css-isubv6 css-1eyhu5m"
>
  Copy
</p>
`;

exports[`Copy shows the correct snapshot for the Copy component tree (tag) 1`] = `
<span
  className="css-6p63zg css-1eyhu5m"
>
  Copy
</span>
`;
